src/cloud/benchflow/sender/sender.go

```diff
@@ -113,15 +118,37 @@ func kafkaConsumer(name string) sarama.PartitionConsumer {
 		}
 	return partConsumer
 }
+*/
+
+func kafkaConsumer(name string) consumergroup.ConsumerGroup {
+	config := consumergroup.NewConfig()
+	config.ClientID = "benchflow"
+	config.Offsets.Initial = sarama.OffsetNewest
+	config.Offsets.ProcessingTimeout = 10 * time.Second
+	consumer, err := consumergroup.JoinConsumerGroup(name+"SparkTasksSenderGroup", []string{name}, []string{kafkaIp+":"+kafkaPort}, config)
+	if err != nil {
+		panic("Could not connect to kafka")
+		}
+	return *consumer
+	}
 
 func consumeFromTopic(t TransformerSetting) {
 	go func() {
 		consumer := kafkaConsumer(t.Topic)
+		cInterruption := make(chan os.Signal, 1)
```

Comment:
In case of abrupt termination it closes the consumer properly, to avoid for example commited offsets from being ignored.